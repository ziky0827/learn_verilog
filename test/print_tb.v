//将输出打印

module print_tb();

    reg [3:0] a;
    reg [3:0] b;
    reg [4:0] d;
    initial begin
    a = 4'ha;
    b = 4'hc;
    d= a+b;
    $display("%d",d);
    end
    reg signed [3:0] data_i;
    reg signed [3:0] data_q;
    reg signed [7:0] data_o;
    initial begin
    data_i=4'sh3;
    data_q=4'shE;
    data_o=$signed(data_i[3:0])*$signed(data_q[3:0]) ;
    $display("%d",data_o);
    end
    
    initial begin
        $display("Hello, World!");
        $finish;
    end
endmodule