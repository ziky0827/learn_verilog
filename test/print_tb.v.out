#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1539-g2693dd32b)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "D:\iverilog\lib\ivl\system.vpi";
:vpi_module "D:\iverilog\lib\ivl\vhdl_sys.vpi";
:vpi_module "D:\iverilog\lib\ivl\vhdl_textio.vpi";
:vpi_module "D:\iverilog\lib\ivl\v2005_math.vpi";
:vpi_module "D:\iverilog\lib\ivl\va_math.vpi";
S_000001c2fbdcee20 .scope module, "print_tb" "print_tb" 2 2;
 .timescale 0 0;
v000001c2fbdc6990_0 .var "a", 3 0;
v000001c2fbdcbde0_0 .var "b", 3 0;
v000001c2fbdcbe80_0 .var "d", 4 0;
v000001c2fbdcbf20_0 .var/s "data_i", 3 0;
v000001c2fbdc6a30_0 .var/s "data_o", 7 0;
v000001c2fbdc6ad0_0 .var/s "data_q", 3 0;
    .scope S_000001c2fbdcee20;
T_0 ;
    %pushi/vec4 10, 0, 4;
    %store/vec4 v000001c2fbdc6990_0, 0, 4;
    %pushi/vec4 12, 0, 4;
    %store/vec4 v000001c2fbdcbde0_0, 0, 4;
    %load/vec4 v000001c2fbdc6990_0;
    %pad/u 5;
    %load/vec4 v000001c2fbdcbde0_0;
    %pad/u 5;
    %add;
    %store/vec4 v000001c2fbdcbe80_0, 0, 5;
    %vpi_call 2 11 "$display", "%d", v000001c2fbdcbe80_0 {0 0 0};
    %end;
    .thread T_0;
    .scope S_000001c2fbdcee20;
T_1 ;
    %pushi/vec4 3, 0, 4;
    %store/vec4 v000001c2fbdcbf20_0, 0, 4;
    %pushi/vec4 14, 0, 4;
    %store/vec4 v000001c2fbdc6ad0_0, 0, 4;
    %load/vec4 v000001c2fbdcbf20_0;
    %pad/s 8;
    %load/vec4 v000001c2fbdc6ad0_0;
    %pad/s 8;
    %mul;
    %store/vec4 v000001c2fbdc6a30_0, 0, 8;
    %vpi_call 2 20 "$display", "%d", v000001c2fbdc6a30_0 {0 0 0};
    %end;
    .thread T_1;
    .scope S_000001c2fbdcee20;
T_2 ;
    %vpi_call 2 24 "$display", "Hello, World!" {0 0 0};
    %vpi_call 2 25 "$finish" {0 0 0};
    %end;
    .thread T_2;
# The file index is used to find the file name in the following table.
:file_names 3;
    "N/A";
    "<interactive>";
    "print_tb.v";
